core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/stringfunction/Concat2Expression.kt

```diff
@@ -1,54 +1,28 @@
 package ch.ergon.dope.resolvable.expression.unaliased.type.function.stringfunction
 
-import ch.ergon.dope.DopeQuery
-import ch.ergon.dope.DopeQueryManager
 import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.FunctionExpression
 import ch.ergon.dope.resolvable.expression.unaliased.type.toDopeType
-import ch.ergon.dope.resolvable.operator.FunctionOperator
 import ch.ergon.dope.validtype.StringType
 
 // Separator is used as an argument and not as a formatting option
 class Concat2Expression(
-    private val separator: TypeExpression<StringType>,
-    private val string: TypeExpression<StringType>,
-    private vararg val strings: TypeExpression<StringType>,
-) : TypeExpression<StringType>, FunctionOperator {
-    override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
-        val separatorDopeQuery = separator.toDopeQuery(manager)
-        val stringDopeQuery = string.toDopeQuery(manager)
-        val stringsDopeQuery = strings.map { it.toDopeQuery(manager) }
-        return DopeQuery(
-            queryString = toFunctionQueryString(
-                symbol = "CONCAT2",
-                separatorDopeQuery,
-                stringDopeQuery,
-                *stringsDopeQuery.toTypedArray(),
-            ),
-            parameters = separatorDopeQuery.parameters + stringDopeQuery.parameters + stringsDopeQuery.fold(
-                emptyMap(),
-            ) { stringParameters, field -> stringParameters + field.parameters },
-        )
-    }
-}
+    separator: TypeExpression<StringType>,
+    string: TypeExpression<StringType>,
+    vararg strings: TypeExpression<StringType>,
+) : FunctionExpression<StringType>("CONCAT2", separator, string, *strings)
 
-fun concat2(separator: TypeExpression<StringType>, string: TypeExpression<StringType>, vararg strings: TypeExpression<StringType>) =
-    Concat2Expression(separator, string, *strings)
+fun concat2(
+    separator: TypeExpression<StringType>,
+    string: TypeExpression<StringType>,
+    vararg strings: TypeExpression<StringType>,
+) = Concat2Expression(separator, string, *strings)
 
-fun concat2(separator: String, string: String, vararg strings: String) = concat2(
-    separator.toDopeType(),
-    string.toDopeType(),
-    *wrapVarargsWithStringValueType(
-        *strings,
-    ),
-)
+fun concat2(separator: String, string: String, vararg strings: String) =
+    Concat2Expression(separator.toDopeType(), string.toDopeType(), *strings.map { it.toDopeType() }.toTypedArray())
 
-fun concat2(separator: TypeExpression<StringType>, string: String, vararg strings: String) = concat2(
-    separator,
-    string.toDopeType(),
-    *wrapVarargsWithStringValueType(
-        *strings,
-    ),
-)
+fun concat2(separator: TypeExpression<StringType>, string: String, vararg strings: String) =
+    Concat2Expression(separator, string.toDopeType(), *strings.map { it.toDopeType() }.toTypedArray())
```

Comment:
It was only used in two files, where as this mapping is used in quite a lot of places. So either we use it everywhere for consistency or leave it out - I chose to leave it out because I thought it doesn't benefit us much. If you think otherwise I can add it again no problem, then I'll just look that it's used everywhere (where feasible of course).